var/spack/repos/builtin/packages/py-onnx-runtime/package.py

```diff
@@ -55,7 +55,7 @@ class PyOnnxRuntime(CMakePackage, PythonExtension):
     # https://github.com/cms-externals/onnxruntime/compare/5bc92df...d594f80
     patch("cms.patch", level=1, when="@1.7.2")
     # https://github.com/cms-externals/onnxruntime/compare/0d9030e...7a6355a
-    patch("cms_1_10.patch", whe="@1.10")
+    patch("cms_1_10.patch", when="@1.10")
```

Comment:
It appears this is a duplicate package: ideally this could be split off into another PR (which removes one of the duplicates and updates dependencies accordingly).

Reply:
> Overall looking good, with a couple of typos. General notes:
> 
> * In several cases this just swaps out an explicit dep on `libiconv` for `iconv` virtual, which seems like a good thing to do in terms of linking consistency (although somewhat a separate concept to the core issue that this PR tackles)

I believe this is necessary for consistency: if `libc` is selected as the required provider for `iconv`, then packages in the DAG with explicit dependencies on `libiconv` could cause problems.

> * I'm wondering if `libintl` should be treated like `iconv` a virtual that may be provided by libc. I think that is the case but it is not modeled that way, so the libiconv/libintl handling differs

I think this was deliberately not done because `gettext` may be required for non-`libintl` reasons. It is perfectly reasonable to use `libc` for `libintl` even in the presence of `gettext` in the DAG.

> As a side note, it would be nice to add a `configure_iconv_prefix_arg` method to all implementations of the `iconv` virtual, which would generally avoid repetition of the following code block:
> 
> ```
>             if spec["iconv"].name == "libc":
>                 args.append("--without-libiconv-prefix")
>             elif not is_system_path(spec["iconv"].prefix):
>                 args.append("--with-libiconv-prefix={p}".format(p=spec["iconv"].prefix))
> ```
> 
> That doesn't have to be done in this PR, but I'll be considering how to refactor this.

This is the `modern` way for the `autotools` `m4` macros to handle this, but it is not the case for older `autotools` versions. In other words, the implementations of `configure_iconv_prefix_arg()` would have to consider `autotools` versions. Even that may not be comprehensive due to package-specific implementations of the handling, but I haven't done a survey.